util/econftool.c

```diff
@@ -0,0 +1,557 @@
+/*
+  Copyright (C) 2019, 2020 SUSE LLC
+  Author: Dominik Gedon <dgedon@suse.de>
+          Jorik Cronenberg <jcronenberg@suse.de>
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+*/
+
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <limits.h>
+#include <ftw.h>
+
+#include "libeconf.h"
+
+static const char *utilname = "econftool";
+static econf_file *key_file = NULL;
+static bool non_interactive = false;
+char *suffix = NULL; /* the suffix of the filename e.g. .conf */
+char path[PATH_MAX]; /* the path of the config file */
+char filename[PATH_MAX]; /* the filename without the suffix */
+char filename_suffix[PATH_MAX]; /* the filename with the suffix */
+char path_filename[PATH_MAX]; /* the path concatenated with the filename and the suffix */
+char root_dir[PATH_MAX] = "/etc";
+char usr_root_dir[PATH_MAX] = "/usr/etc";
```

Comment:
Variables which aren't written to should be `const`, especially global ones.
Please also make them static.

For strings, you can omit the length, otherwise they'll be padded with zeros:
`static const char usr_root_dir[] = "/usr/etc";`

Reply:
They are being written to in line 529 and 530:
```
    change_root_dir(root_dir);
    change_root_dir(usr_root_dir);
```